src/Microdown/MicHTTPResource.class.st

```diff
@@ -34,16 +35,24 @@ MicHTTPResource class >> imageCache [
 ]
 
 { #category : #images }
+MicHTTPResource class >> imageNotCachedBlock: aHandlerBlock [ 
+	"If an image is not in the cache, execute this block get an image. 
+	Do not store the produced image in the cache."
+	ImageNotCachedBlock := aHandlerBlock
+]
+
+{ #category : #'class initialization' }
 MicHTTPResource class >> initialize [
 	<script>
-	ImageCache := nil
+	ImageCache := nil.
+	ImageNotCachedBlock := nil
 ]
 
 { #category : #images }
 MicHTTPResource class >> rawImageFrom: aURL [
 	[ ^ ZnEasy getImageOfType: (ZnMimeType main: 'image' sub: '*') fromUrl: aURL ]
-	on: Error
-	do: [ ^nil ]
+	on: Exception
+	do: [ ^ nil ]
```

Comment:
I was thinking that we could just return an image so that the method always returns polymorphic objects.

Reply:
The nil is used to let the renderer rather than the ressource decide what to do when no image is found. I believe this is the best way.

Reply:
I am not convinced the polymorphism should sit here. I will leave it as is.